configure.ac

```diff
@@ -3165,12 +3165,45 @@ if test "$have_glibc_memmove_bug" = yes; then
     AS_VAR_APPEND([LIBMPDEC_CFLAGS], [" -U_FORTIFY_SOURCE"])
 fi
 
+dnl Check for SQLite library. Use pkg-config if available.
+PKG_CHECK_MODULES(
+  [LIBSQLITE3], [sqlite3 >= 3.7.15], [], [
+    LIBSQLITE3_LIBS="-lsqlite3"
+    LIBSQLITE3_CFLAGS=
+  ]
+)
+AS_VAR_APPEND([LIBSQLITE3_CFLAGS], [' -I$(srcdir)/Modules/_sqlite'])
+
+AS_VAR_COPY([save_CFLAGS], [CFLAGS])
+AS_VAR_COPY([save_LDFLAGS], [LDFLAGS])
+AS_VAR_COPY([CFLAGS], [LIBSQLITE3_CFLAGS])
+AS_VAR_COPY([LDFLAGS], [LIBSQLITE3_LIBS])
```

Comment:
Idea for another PR: We push and pop LIBS, CFLAGS, and LDFLAGS pretty often. How do you feel about two macros to automate the task?

Reply:
Yes! I've been looking for something like that in the autoconf docs, and it surprises me that it does not exist. It seems to be a very common operation. I think I would prefer a push/pop kind of naming.

Reply:
I opened GH-29637 for this. Let me know what you think.